### ISSUE TITLE
deepLinkHandler callback not called

### ISSUE THREAD

**Issue by mlazari** [BranchMetrics/Segment-Branch-iOS#14]
Hello. Is it possible to use this integration and also use a deepLinkHandler callback?

I have this in `application:didFinishLaunchingWithOptions:`:
```
[[Branch getInstance] initSessionWithLaunchOptions:launchOptions andRegisterDeepLinkHandler:^(NSDictionary *params, NSError *error) {
  // This is called when not using this integration, but not calling when using it
}];
```
I think the issue happens because the integration reinitialize branch session here without deepLinkHandler callback and my deepLinkHandler is lost after that (the deepLinkHandler is called if I remove the first line from this method):
```
- (void)applicationDidFinishLaunching:(NSNotification *)notification {
    [[Branch getInstance] initSessionWithLaunchOptions:notification.userInfo];
    SEGLog(@"[[Branch getInstance] initSessionWithLaunchOptions:%@]", notification.userInfo);
}
```

Xcode shows a warning for the initSessionWithLaunchOptions method used here:

> This is not the recommended method of initializing Branch. While Branch is able to properly attribute deep linking info with the launch options, you lose the ability to do anything with a callback.

**Comment by E-B-Smith** [comment]
You don't need to pass the launch options to Branch when you're using the Segment integration.

If you'd like to set a deep link callback block on Branch, you can set it directly like this:

```
[Branch getInstance].sessionInitWithParamsCallback = ^(NSDictionary *params, NSError *error) {
    ...
};
```

Note that this is available in Branch versions 0.25.4 and up, which will be released today.

Also, consider using NSNotifications to get deep link callbacks.  This is available in the SDK now.

First register for the notification:

https://github.com/BranchMetrics/Segment-Branch-iOS/blob/master/Examples/Fortune/Fortune/FortuneViewController.swift#L33-L38

And then handle the notification:

https://github.com/BranchMetrics/Segment-Branch-iOS/blob/master/Examples/Fortune/Fortune/FortuneViewController.swift#L101

**Comment by mlazari** [comment]
Thank you for looking into this. I thought about setting the callback without calling initSession method (because I've seen that the integration already called a initSession method), but I didn't find a way to do it in current version.
Where should I set the callback? I think it should happen after the call of `[[Branch getInstance] initSessionWithLaunchOptions:notification.userInfo];` in Segment-Branch-iOS. Otherwise this call will re-set it to null: https://github.com/BranchMetrics/ios-branch-deep-linking/blob/5dd137d3e224b7f0ddc7c3b4906c3a2f6a380e51/Branch-SDK/Branch-SDK/Branch.m#L563

I think I'll try to use NSNotifications instead, as you suggested.

L.E. NSNotification works, thank you.
